frontend/src/hooks/useTimeout.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useRef } from 'react'
+
+export const useTimeout = (
+  callback: (...args: unknown[]) => void,
+  delay: number | null,
```

Comment:
`null` is required so we can cancel the timeout if it is not supplied. This hook will be reinvoked whenever the `delay` value changes

Reply:
Wait(sorry I'm confused now 😅 ), how does invoking with `delay:null` cancels the timeout?

Reply:
Hehe, no worries. It's some abstraction of `useEffect` that trips up people new to React (and why some ppl hate hooks)

When `delay` changes, the `useEffect` block in line 9 re-runs, since `delay` is a dependency of the effect (line 23).
Before useEffect re-runs, it will always call the cleanup function that is declared in the return callback of the useEffect hook (line 18). This means `clearTimeout` will be invoked when `delay` is changed (to `null`, or even in fact anything as long as it differs from the current `delay` value).

See more at https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1

Reply:
Got it! Thank you Sir 🙏